src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -202,11 +205,37 @@ public function supportsDenormalization($data, $type, $format = null, array $con
      */
     private function getNormalizer($data, $format, array $context)
     {
+        $type = \is_object($data) ? 'c-'.\get_class($data) : \gettype($data);
+        if (
+            isset($this->normalizerCache[$type][$format]) ||
+            (isset($this->normalizerCache[$type]) && \array_key_exists($format, $this->normalizerCache[$type]))
```

Comment:
`if (isset($this->normalizerCache[$type][[$format]))` looks enough to me here, provided the case for primitive types is dont this way:
- `$this->normalizerCache[$type][$format] = false;`
- \return $this->normalizerCache[$type][$format] ?: null;`